modules/GenericSensor.js

```diff
@@ -126,8 +127,13 @@ function GenericSensor() {
                            sensor.z !== null,
                            "sensor: reading value for '" + sensorType + "'");
 
-                    assert(sensor.controller === "bmi160",
-                           "sensor: controller name as default");
+                    if (sensorType === "Magnetometer") {
```

Comment:
I think this was discussed in #1055, I don't think we should be checking the sensor.controller as it's not a property that will be part of the API.  The controller is only passed as an optional argument to the constructor, and once the object is created, it's no longer relevant.  @cuiyanx so can you maybe explain what is the need here to check the controller value after it has been set?  Maybe we need to update the API doc a little better to not confuse the developers.